lib/ts/unit-tests/test_layout.cpp

```diff
@@ -0,0 +1,108 @@
+/** @file
+  Test file for layout structure
+  @section license License
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+
+#include "catch.hpp"
+
+#include "ts/string_view.h"
+#include "ts/I_Layout.h"
+#include "ts/ink_platform.h"
+#include <string>
+
+using namespace std;
+
+void
+append_slash(string &path)
+{
+  if (path.back() != '/')
+    path.append("/");
+}
+
+// test cases: [constructor], [env_constructor], [create], [relative], [relative_to], [update_sysconfdir]
+// ======= test for layout ========
+
+TEST_CASE("constructor test", "[constructor]")
+{
+  Layout layout;
+  // test for constructor
+  REQUIRE(layout.prefix == TS_BUILD_PREFIX);
+  REQUIRE(layout.sysconfdir == layout.relative(TS_BUILD_SYSCONFDIR));
+}
+
+TEST_CASE("environment variable constructor test", "[env_constructor]")
+{
+  string newpath = TS_BUILD_PREFIX;
+  append_slash(newpath);
+  newpath += "env";
```

Comment:
sometimes we don't know if the path contains the end slash or not. append_slash check if we already have slash and then append it. I'll make this function look better by returning the string and pass in the path as const.